src/Packagist/WebBundle/Package/Updater.php

```diff
@@ -629,11 +629,12 @@ private function prepareReadme($readme, $isGithub = false, $owner = null, $repo
             }
         }
 
-        // remove first title as it's usually the project name which we don't need
-        if ($dom->getElementsByTagName('h1')->length) {
+        // remove first page element if it's a <h1> or <h2>, because it's usually
+        // the project name or the `README` string which we don't need
+        if ('h1' === $dom->childNodes->item(0)->nodeName) {
             $first = $dom->getElementsByTagName('h1')->item(0);
```

Comment:
could be simplified:

```php
$first = $dom->childNodes->item(0);

if ($first && ('h1' === $first->nodeName || 'h2' === $first->nodeName)) {
    $first->parentNode->removeChild($first);
}
```

My snippet also account for the case of empty readme, where there could be no first node.